libraries/lib-uuid/Uuid.cpp

```diff
@@ -0,0 +1,216 @@
+/*!********************************************************************
+
+ Audacity: A Digital Audio Editor
+
+ @file Uuid.cpp
+ @brief Define a class to generate and parse UUIDs
+
+ Dmitry Vedenko
+ **********************************************************************/
+
+/*!********************************************************************
+
+ @class Uuid
+ @brief Platform independent class for generating and parsing UUIDs.
+
+ **********************************************************************/
+
+#include "Uuid.h"
+
+#include <algorithm>
+#include <cstring>
+
+#if defined(USE_UUID_CREATE)
+#   include <rpc.h>
+#elif defined(USE_CFUUID)
+#   include <CoreFoundation/CFUUID.h>
+#elif defined(USE_LIBUUID)
+#   include <uuid/uuid.h>
+#endif
+
+#include "HexHelpers.h"
+
+namespace audacity
+{
+
+namespace
+{
+bool readByte(Uuid::Bytes::iterator& outputIt,
+              std::string::const_iterator& inputIt,
+              const std::string::const_iterator& inputEnd)
+{
+   if (inputIt == inputEnd)
+      return false;
+
+   const char c1 = *inputIt++;
+
+   if (!std::isxdigit(c1))
+      return false;
+
+   if (inputIt == inputEnd)
+      return false;
+
+   const char c2 = *inputIt++;
+
+   if (!std::isxdigit(c2))
+      return false;
+
+   *outputIt = uint8_t((HexCharToNum(c1) << 4) | HexCharToNum(c2));
```

Comment:
Let's avoid C-style casts or function notation for scalar types.

Either use uniform initialization where you can:
'uint8_t{...}'

Or if you can't (because UI doesn't allow narrowing; if HexCharToNum returned int for instance), then use static_cast.

Reply:
I lately came across similar concerns preparing my updated less-dithering branch.

Reply:
Well, actually it is a C++ cast using converting constructor or whatever it's called. I can change it to static_cast, although this form is shorter and more expressive.

The cast itself is required because of integer promotion rules when performing <<

Reply:
I try to be alert to truncation problems.  Sure, there is no harm here.

So actually it's just the first operand of | that needs a static_cast.